easybuild/tools/module_generator.py

```diff
@@ -407,6 +407,29 @@ def det_installdir(self, modfile):
 
         return res
 
+    def unpack_setenv_value(self, env_var_name, env_var_val):
+        """
+        Unpack value that specifies how to define an environment variable with specified name.
+        """
+        use_pushenv = False
+
+        # value may be specified as a string, or as a dict for special cases
+        if isinstance(env_var_val, string_type):
+            value = env_var_val
+
+        elif isinstance(env_var_val, dict):
+            use_pushenv = env_var_val.get('pushenv', False)
+            try:
+                value = env_var_val['value']
+            except KeyError:
+                raise EasyBuildError("Required key 'value' is missing in dict that specifies how to set $%s: %s",
+                                     env_var_name, env_var_val)
+        else:
+            raise EasyBuildError("Incorrect value type for setting $%s environment variable (%s): %s",
```

Comment:
Should this include list of valid types?

Reply:
Yes, that would be helpful indeed, I'll change that, thanks for the suggestion!